apps/aecontract/src/aect_state_tree.erl

```diff
@@ -149,18 +149,67 @@ root_hash(#contract_tree{contracts = CtTree}) ->
                      {'ok', binary(), aec_poi:poi()}
                    | {'error', 'not_present' | 'wrong_root_hash'}.
 add_poi(Id, #contract_tree{contracts = CtTree}, Poi) ->
-    aec_poi:add_poi(Id, CtTree, Poi).
+    case aec_poi:add_poi(Id, CtTree, Poi) of
+        {ok, Contract, ContractPoi} ->
+            add_store_to_poi(
+              aect_contracts:deserialize(Id, Contract),
+              CtTree,
+              ContractPoi);
+        {error, _} = Error -> Error
+    end.
+
+add_store_to_poi(Contract, CtTree, Poi) ->
+    Id = aect_contracts:store_id(Contract),
+    Iterator = aeu_mtrees:iterator_from(Id, CtTree),
+    Next = aeu_mtrees:iterator_next(Iterator),
+    Size = byte_size(Id),
+    case add_store_keys_poi(Id, Next, Size, #{}, Poi, CtTree) of
+        {error, _} = E -> E;
+        {Store, StorePoi} ->
+            {ok, aect_contracts:serialize_for_poi(aect_contracts:set_state(Store, Contract)), StorePoi}
+    end.
+
+add_store_keys_poi(_, '$end_of_table', _, Store, Poi, _) ->
+    {Store, Poi};
+add_store_keys_poi(Id, {PrefixedKey, Val, Iter}, PrefixSize, Store, Poi, CtTree) ->
+    case PrefixedKey of
+        <<Id:PrefixSize/binary, Key/binary>> = KeyId ->
+            Store1 = Store#{ Key => Val},
+            case aec_poi:add_poi(KeyId, CtTree, Poi) of
```

Comment:
I see this line is enriching the PoI proof DB for the contracts tree with the contract store item (key value). Ok.